src/formatDate.js

```diff
@@ -9,6 +9,58 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const splittedDate = date.split(fromFormat[3]);
+  const oldDateObj = {};
+  const result = [];
+
+  for (let i = 0; i < splittedDate.length; i++) {
+    oldDateObj[fromFormat[i]] = splittedDate[i];
+    // make new object via input date;
+  }
+  changeYearsFormat(oldDateObj, toFormat);
```

Comment:
The function `changeYearsFormat` mutates its first argument `oldDateObj`. This is not evident from the function's name or its usage here. It's better to avoid such side effects as they can lead to harder-to-understand code.